products/datastore/api.yaml

```
@@ -0,0 +1,94 @@
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+--- !ruby/object:Api::Product
+name: Datastore
+display_name: Cloud Datastore
+versions:
+  - !ruby/object:Api::Product::Version
+    name: ga
+    base_url: https://datastore.googleapis.com/v1/
+scopes:
+  - https://www.googleapis.com/auth/datastore
+apis_required:
+  - !ruby/object:Api::Product::ApiReference
+    name: Cloud Datastore APU
+    url: https://console.cloud.google.com/apis/library/datastore.googleapis.com
+async: !ruby/object:Api::OpAsync
+  operation: !ruby/object:Api::OpAsync::Operation
+    path: 'name'
+    base_url: '{{op_id}}'
+    wait_ms: 1000
+  result: !ruby/object:Api::OpAsync::Result
+    path: 'response'
+    resource_inside_response: true
+  status: !ruby/object:Api::OpAsync::Status
+    path: 'done'
+    complete: true
+    allowed:
+      - true
+      - false
+  error: !ruby/object:Api::OpAsync::Error
+    path: 'error'
+    message: 'message'
+
+objects:
+  - !ruby/object:Api::Resource
+    name: 'Index'
+    base_url: "projects/{{project}}/indexes"
+    self_link: "projects/{{project}}/indexes/{{indexId}}"
+    input: true
+    identity:
+      - indexId
+    description: |
+      Describes a composite index for Cloud Datastore.
+    properties:
+      - !ruby/object:Api::Type::String
+        name: 'indexId'
+        output: true
+        description: |
+          The index id. Output only.
+      - !ruby/object:Api::Type::String
+        name: 'kind'
+        required: true
+        description: |
+          The entity kind which the index applies to.
+      - !ruby/object:Api::Type::Enum
+        name: 'ancestor'
+        default_value: :NONE
+        values:
+          - :NONE
+          - :ALL_ANCESTORS
+        description: |
+            Policy for including ancestors in the index.  Either all ancestors or none will
+            be included, the default is none.
+      - !ruby/object:Api::Type::Array
+        name: 'properties'
+        description: |
+            An ordered list of properties to index on.
+        min_size: 1
+        item_type: !ruby/object:Api::Type::NestedObject
+          properties:
+            - !ruby/object:Api::Type::String
+              name: 'name'
+              required: true
+              description: |
+                  The property name to index.
+            - !ruby/object:Api::Type::Enum
+              name: 'direction'
+              required: true
+              values:
+                - :ASCENDING
+                - :DESCENDING
+              description: |
+                 Which direction should the index optimize for sorting?
```

nit: I prefer docs to be phrased as statements, not questions, but up to you
```suggestion
                 The direction the index should optimize for sorting. Possible values are ASCENDING and DESCENDING.
```